src/client/file_instance.cpp

```diff
@@ -168,11 +176,13 @@ int FileInstance::AioDiscard(CurveAioContext* aioctx) {
 //    再去打开，所以这时候需要获取mds一侧session lease时长，然后在client这一侧
 //    等待一段时间再去Open，如果依然失败，就向上层返回失败。
 int FileInstance::Open(const std::string& filename,
-                       const UserInfo& userinfo,
-                       std::string* sessionId) {
+                        const UserInfo& userinfo,
+                        const int curveOpenFlags,
+                        std::string* sessionId) {
     LeaseSession_t  lease;
     int ret = LIBCURVE_ERROR::FAILED;
-
+    finfo_.context.uuid = curve::common::UUIDGenerator().GenerateUUID();
+    finfo_.context.openflags = curveOpenFlags;
```

Comment:
you should check openflags, because it has to conform to certain rules, like it should either `CURVE_EXCLUSIVE` or `CURVE_SHARED`, and should not open with CURVE_RDWR and CURVE_RDONLY.